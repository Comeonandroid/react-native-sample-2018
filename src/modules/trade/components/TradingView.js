import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { WebView } from 'react-native'

class TradingView extends PureComponent {
  render() {
    const { url } = this.props
    return (
      <Web source={{ uri: url }} />
    )
  }
}

TradingView.propTypes = {
  url: PropTypes.string
}

const Web = styled(WebView)`
  flex: 1;
  width: 100%;
`

export default TradingView
