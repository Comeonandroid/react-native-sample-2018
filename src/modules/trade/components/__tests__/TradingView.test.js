import 'react-native'
import React from 'react'

import { itBehavesLike } from '../../../../specs/shared_examples/index'

import TradingView from '../TradingView'

const setup = propOverrides => {
  const props = Object.assign({
    url: 'https://s3-ap-southeast-1.amazonaws.com/tokenjar/mobile_white.html'
  }, propOverrides)

  const component = <TradingView {...props} />

  return {
    props,
    component
  }
}

describe('<TradingView />', () => {
  const { component } = setup()
  itBehavesLike('aThemedComponent', { subject: component })
})