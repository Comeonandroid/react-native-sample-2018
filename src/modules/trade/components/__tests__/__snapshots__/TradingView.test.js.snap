// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TradingView /> dark: renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTWebView
    allowsInlineMediaPlayback={undefined}
    automaticallyAdjustContentInsets={undefined}
    bounces={undefined}
    contentInset={undefined}
    dataDetectorTypes={undefined}
    decelerationRate={undefined}
    injectedJavaScript={undefined}
    mediaPlaybackRequiresUserAction={undefined}
    messagingEnabled={false}
    onLoadingError={[Function]}
    onLoadingFinish={[Function]}
    onLoadingStart={[Function]}
    onMessage={[Function]}
    onShouldStartLoadWithRequest={undefined}
    scalesPageToFit={true}
    scrollEnabled={undefined}
    source={
      Object {
        "uri": "https://s3-ap-southeast-1.amazonaws.com/tokenjar/mobile_white.html",
      }
    }
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "width": "100%",
          },
          undefined,
        ],
      ]
    }
  />
</View>
`;

exports[`<TradingView /> light: renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTWebView
    allowsInlineMediaPlayback={undefined}
    automaticallyAdjustContentInsets={undefined}
    bounces={undefined}
    contentInset={undefined}
    dataDetectorTypes={undefined}
    decelerationRate={undefined}
    injectedJavaScript={undefined}
    mediaPlaybackRequiresUserAction={undefined}
    messagingEnabled={false}
    onLoadingError={[Function]}
    onLoadingFinish={[Function]}
    onLoadingStart={[Function]}
    onMessage={[Function]}
    onShouldStartLoadWithRequest={undefined}
    scalesPageToFit={true}
    scrollEnabled={undefined}
    source={
      Object {
        "uri": "https://s3-ap-southeast-1.amazonaws.com/tokenjar/mobile_white.html",
      }
    }
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "width": "100%",
          },
          undefined,
        ],
      ]
    }
  />
</View>
`;
